docker/lwua-ingest/lwua-py/tests/test_queries.py

```diff
@@ -0,0 +1,107 @@
+# test file to test the JRDF builder and all the templates
+
+from pyrdfj2 import J2RDFSyntaxBuilder
+from lwua.helpers import resolve_path
+import os
+
+URN_BASE = os.getenv("URN_BASE", "urn:lwua:INGEST")
+
+
+def get_j2rdf_builder():
+    template_folder = os.path.join(
+        os.path.dirname(__file__),
+        "../lwua/templates")
+    # init J2RDFSyntaxBuilder
+    context = f"{URN_BASE}:ADMIN"
+    j2rdf = J2RDFSyntaxBuilder(
+        templates_folder=template_folder,
+        extra_functions={"registry_of_lastmod_context": context},
+    )
+    return j2rdf
+
+
+J2RDF = get_j2rdf_builder()
+
+
+def test_template_insert_graph():
+    # Arrange
+    template = "insert_graph.sparql"
+    vars = {
+        "context": "urn:lwua:INGEST:test_file.txt",
+        "raw_triples": "<http://example.com/subject> <http://example.com/predicate> <http://example.com/object> .",
+    }
+
+    # Act
+    query = J2RDF.build_syntax(template, **vars)
+
+    # clean up the query by removing the newlines
+    query = query.replace("\n", "")
+
+    print(query)
+    to_expect = "INSERT DATA { GRAPH <urn:lwua:INGEST:test_file.txt> { <http://example.com/subject> <http://example.com/predicate> <http://example.com/object> . } }"
+    # Assert
+    assert query == to_expect, f"Expected '{to_expect}', but got '{query}'"
```

Comment:
unsure -- but if RDFlib also supports insert statements onto an in-memory graph -- then you could actually provide a mock_gdb variable that allows for further testing